shared/src/main/scala/mlscript/syntax.scala

```diff
@@ -94,8 +94,13 @@ sealed abstract class NullaryType                        extends Type
 
 case object Top                                          extends NullaryType
 case object Bot                                          extends NullaryType
+
+// Literal type or singleton type i.e.
+// `type t: 0` is a type with only one possible value
+// which is the literal integer 0 itself
 final case class Literal(lit: Lit)                       extends NullaryType
 
+// declare a new type with a given name
```

Comment:
This is not about declaring types; it's about referencing them.

Reply:
Fixed